addons/stock/models/product.py

```diff
@@ -97,6 +97,17 @@ class Product(models.Model):
         compute='_compute_nbr_reordering_rules', compute_sudo=False)
     putaway_rule_ids = fields.One2many('stock.putaway.rule', 'product_id', 'Putaway Rules')
     storage_category_capacity_ids = fields.One2many('stock.storage.category.capacity', 'product_id', 'Storage Category Capacity')
+    show_on_hand_qty_status_button = fields.Boolean(compute='_compute_show_qty_status_button')
+    show_forecasted_qty_status_button = fields.Boolean(compute='_compute_show_qty_status_button')
+
+    def _compute_show_qty_status_button(self):
+        for product in self:
+            if product.type != 'product':
+                product.show_on_hand_qty_status_button = False
+                product.show_forecasted_qty_status_button = False
+            else:
+                product.show_on_hand_qty_status_button = True
+                product.show_forecasted_qty_status_button = True
```

Comment:
No need to define it on product.product. On product.template it should be enough I think

Reply:
My bad, some products could be kit and other not...

Reply:
Or you can remove it there and only redefine it in mrp

Reply:
no, it has to be in stock. we have sale_stock_renting also modified button attrs
https://github.com/odoo/enterprise/pull/20440/commits/5c72a753e347190189332f957dd7b147d4af3cd8

Reply:
Gosh

Reply:
Then do the compute base on the product_tmpl_id
```python
product.show_on_hand_qty_status_button = product.product_tmpl_id.show_on_hand_qty_status_button
product.show_forecasted_qty_status_button = product.product_tmpl_id.show_forecasted_qty_status_button
```